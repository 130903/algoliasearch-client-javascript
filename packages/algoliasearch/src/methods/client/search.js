// @flow
import type { IndexName, SearchParameters } from 'algoliasearch';
import type { RequestMethod, RequestOptions } from 'algoliasearch-requester';

export type SearchRequest = {|
  indexName: IndexName,
  params: SearchParameters,
|};

export default function search(
  { requests = [] }: { requests: SearchRequest[] },
  {
    requester,
    requestOptions,
  }: {
    requester: RequestMethod,
    requestOptions?: RequestOptions,
  }
) {
  return requester({
    method: 'POST',
    path: '/1/indexes/*/queries',
    body: { requests },
    requestOptions,
    requestType: 'read',
  });
}
