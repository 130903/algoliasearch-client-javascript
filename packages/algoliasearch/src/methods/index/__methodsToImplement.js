// index methods that should be implemented, `true` if they are

const apiMethods = {
  addApiKey: false,
  addObject: false,
  addObjects: false,
  batch: true,
  batchRules: false,
  batchSynonyms: false,
  browse: true,
  browseAll: false,
  browseFrom: true,
  clearRules: false,
  clearSynonyms: false,
  deleteApiKey: false,
  deleteByQuery: false,
  deleteObject: false,
  deleteObjects: false,
  deleteRule: false,
  deleteSynonym: false,
  getApiKey: false,
  getObject: true,
  getObjects: true,
  getRule: false,
  getSettings: false,
  getSynonym: false,
  listApiKeys: false,
  partialUpdateObject: false,
  partialUpdateObjects: false,
  saveObject: false,
  saveObjects: false,
  saveRule: false,
  saveSynonym: false,
  search: true,
  searchForFacetValues: false,
  searchRules: false,
  searchSynonyms: false,
  setSettings: false,
  similarSearch: false,
  updateApiKey: false,
  waitTask: true,
};

const requesterMethods = {
  clearCache: false,
};
