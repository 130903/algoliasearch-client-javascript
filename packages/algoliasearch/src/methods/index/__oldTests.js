import { initIndex } from './';

const validParams = {
  appId: 'some_id',
  apiKey: 'some_key',
  indexName: 'some_index',
};

const snapshotAll = requests =>
  requests.map(req => req.then(sn => expect(sn).toMatchSnapshot()));

it('batch', () => {
  const index = initIndex(validParams);
  const requests = [
    index.batch([
      {
        action: 'addObject',
        body: {
          some: 'data',
          nested: { is: { all: 'fine' } },
          function: is => `${is}also allowed, but not smart`,
        },
      },
    ]),
    index.batch([
      {
        action: 'updateObject',
        body: {
          objectID: 'some-cool-object',
          some: 'other data',
          and: 'this is everything now',
        },
      },
    ]),
    index.batch([
      {
        action: 'partialUpdateObject',
        body: {
          objectID: 'some-cool-object',
          some: 'data',
          the: 'rest is still here',
        },
      },
    ]),
    index.batch([
      {
        action: 'partialUpdateObjectNoCreate',
        body: {
          objectID: 'some-cool-object',
          this: 'object',
          "won't": 'be made',
          if: "it doesn't exist yet",
        },
      },
    ]),
    index.batch([
      {
        action: 'deleteObject',
        body: {
          objectID: 'some-uncool-object',
        },
      },
    ]),
    index.batch([
      {
        action: 'delete',
      },
    ]),
    index.batch([
      {
        action: 'clear',
      },
    ]),
  ];

  snapshotAll(requests);
});

it('clear', () => {
  const index = initIndex(validParams);

  const requests = [index.clear()];
  snapshotAll(requests);
});

it('copy', () => {
  const index = initIndex(validParams);

  const requests = [index.copy('new_index')];
  snapshotAll(requests);
});

it('remove', () => {
  const index = initIndex(validParams);

  const requests = [index.remove()];
  snapshotAll(requests);
});

it('move', () => {
  const index = initIndex(validParams);

  const requests = [index.move('new_index')];
  snapshotAll(requests);
});

it('browse', () => {
  const index = initIndex(validParams);

  const requests = [index.browse(), index.browse({ hitsPerPage: 40 })];

  snapshotAll(requests);
});

it('browseFrom', () => {
  const index = initIndex(validParams);

  const requests = [index.browseFrom('some_weirdo_cursor')];
  snapshotAll(requests);
});

it('waitTask', () => {
  const index = initIndex(validParams);
  const requests = [index.waitTask('some_task_id')];

  snapshotAll(requests);
});
