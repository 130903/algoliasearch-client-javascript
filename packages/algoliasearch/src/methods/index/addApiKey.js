// @flow

import type { IndexName } from 'algoliasearch';
import type { RequestMethod, RequestOptions } from 'algoliasearch-requester';

export type ACL =
  | 'search'
  | 'browse'
  | 'addObject'
  | 'deleteObject'
  | 'deleteIndex'
  | 'settings'
  | 'editSettings'
  | 'analytics'
  | 'listIndexes';

export type Parameters = {|
  acls?: ACL[],
  description?: string,
  maxHitsPerQuery?: number,
  maxQueriesPerIPPerHour?: number,
  queryParameters?: string,
  referers?: string[],
  validity?: number,
|};

export default function addApiKey(
  body: Parameters,
  {
    requester,
    indexName,
    requestOptions,
  }: {
    requester: RequestMethod,
    indexName: IndexName,
    requestOptions?: RequestOptions,
  }
) {
  return requester({
    method: 'POST',
    path: `/1/indexes/${indexName}/keys/`,
    body,
    requestOptions,
    requestType: 'write',
  });
}
