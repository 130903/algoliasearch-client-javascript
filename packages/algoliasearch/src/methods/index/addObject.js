// @flow

import type { RequestMethod, IndexName } from '../../types';

export default function addObject(
  req: RequestMethod,
  indexName: IndexName,
  body: Object
) {
  return req({
    method: 'POST',
    path: `/1/indexes/${indexName}/`,
    body,
  });
}
