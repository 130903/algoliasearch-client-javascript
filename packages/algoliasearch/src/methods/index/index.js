// @flow

export { default as addApiKey } from './addApiKey';
export { default as addObject } from './addObject';
export { default as addObjects } from './addObjects';
export { default as batch } from './batch';
export { default as batchRules } from './batchRules';
export { default as batchSynonyms } from './batchSynonyms';
export { default as browse } from './browse';
export { default as browseAll } from './browseAll';
export { default as browseFrom } from './browseFrom';
export { default as clearRules } from './clearRules';
export { default as clearSynonyms } from './clearSynonyms';
export { default as deleteApiKey } from './deleteApiKey';
export { default as deleteBy } from './deleteBy';
export { default as deleteObject } from './deleteObject';
export { default as deleteObjects } from './deleteObjects';
export { default as deleteRule } from './deleteRule';
export { default as deleteSynonym } from './deleteSynonym';
export { default as getApiKey } from './getApiKey';
export { default as getObject } from './getObject';
export { default as getObjects } from './getObjects';
export { default as getRule } from './getRule';
export { default as getSettings } from './getSettings';
export { default as getSynonym } from './getSynonym';
export { default as index } from './index';
export { default as listApiKeys } from './listApiKeys';
export { default as partialUpdateObject } from './partialUpdateObject';
export { default as partialUpdateObjects } from './partialUpdateObjects';
export { default as saveObject } from './saveObject';
export { default as saveObjects } from './saveObjects';
export { default as saveRule } from './saveRule';
export { default as saveSynonym } from './saveSynonym';
export { default as search } from './search';
export { default as searchForFacetValues } from './searchForFacetValues';
export { default as searchRules } from './searchRules';
export { default as searchSynonyms } from './searchSynonyms';
export { default as setSettings } from './setSettings';
export { default as similarSearch } from './similarSearch';
export { default as updateApiKey } from './updateApiKey';
export { default as waitForCompletion } from './waitForCompletion';
