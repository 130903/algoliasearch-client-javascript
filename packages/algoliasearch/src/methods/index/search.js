// @flow
import type { IndexName, SearchParameters } from 'algoliasearch';
import type { RequestMethod, RequestOptions } from 'algoliasearch-requester';

export default function search(
  params: SearchParameters = {},
  {
    requester,
    indexName,
    requestOptions,
  }: {
    requester: RequestMethod,
    indexName: IndexName,
    requestOptions?: RequestOptions,
  }
) {
  return requester({
    method: 'POST',
    path: `/1/indexes/${indexName}/query`,
    body: { params },
    requestOptions,
    requestType: 'read',
  });
}
