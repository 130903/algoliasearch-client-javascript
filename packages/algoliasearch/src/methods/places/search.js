// @flow
import type { RequestMethod, SearchParameters } from '../../types';

export default function search(req: RequestMethod, params: SearchParameters) {
  return req({
    method: 'POST',
    path: `/1/indexes/places/query`,
    body: { params },
  });
}
