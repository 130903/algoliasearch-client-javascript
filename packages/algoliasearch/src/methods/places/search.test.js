import { snapshotAll } from '../../testUtils/index.js';
import requester from 'algoliasearch-http-requester';
import search from './search.js';

it('search', () => {
  const requests = [
    search({}, { requester }),
    search({ hitsPerPage: 4 }, { requester }),
  ];
  snapshotAll(requests);
});
