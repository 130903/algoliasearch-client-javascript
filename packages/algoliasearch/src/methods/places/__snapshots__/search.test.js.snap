// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search 1`] = `
Object {
  "body": Object {
    "params": Object {},
  },
  "method": "POST",
  "path": "/1/indexes/places/query",
  "requestOptions": undefined,
  "requestType": "read",
}
`;

exports[`search 2`] = `
Object {
  "body": Object {
    "params": Object {
      "hitsPerPage": 4,
    },
  },
  "method": "POST",
  "path": "/1/indexes/places/query",
  "requestOptions": undefined,
  "requestType": "read",
}
`;
